src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -218,39 +227,121 @@ class ActionDelegate {
      * @param field which behaviour will be changed
      * @param behavior one of visible, editable, required, optional, hidden, forbidden
      */
     def make(Field field, Closure behavior) {
-        [on: { Transition trans ->
-            [when: { Closure condition ->
-                if (condition()) {
-                    behavior(field, trans)
-                    ChangedField changedField = new ChangedField(field.stringId)
-                    changedField.addAttribute("type", field.type.name)
-                    changedField.addBehavior(useCase.dataSet.get(field.stringId).behavior)
-                    SetDataEventOutcome outcome = createSetDataEventOutcome()
-                    outcome.addChangedField(field.stringId, changedField)
-                    this.outcomes.add(outcome)
+        [on: { Object optionalTransition ->
+            if (optionalTransition instanceof Transition) {
+                [when: { Closure condition ->
+                    if (condition()) {
+                        behavior(field, optionalTransition)
+                        saveFieldBehavior(field, optionalTransition)
+                    }
+                }]
+            } else if (optionalTransition instanceof Closure) {
+                if (optionalTransition == each) {
+                    useCase.petriNet.transitions.each {transitionEntry ->
+                        Transition trans = transitionEntry.value
+                        if (trans.dataSet.containsKey(field.stringId)) {
+                            behavior(field, trans)
+                            saveFieldBehavior(field, trans)
+                        }
+                    }
                 }
-            }]
+            } else {
+                throw new IllegalArgumentException("Invalid call of make method. Method call should contain transition or keyword \'each\'.")
+            }
+        }]
+    }
+
+    /**
+     * Changes behavior of a given fields on given transition or on each transition containing a field.
+     * <br>
+     * Example 1:
+     * <pre>
+     *     text: f.textId,
+     *     otherText: f.otherTextId,
+     *     transition: t.transitionId;
+     *
+     *     if (condition)
+     *          make [text, otherText], visible on transition
+     * </pre>
+     * This code will change the behavior of fields <i>text</i> and <i>otherText</i> to <i>visible</i> on given transition.
+     *
+     * Example 2:
+     * <pre>
+     *      text: f.textId,
+     *      otherText: f.otherTextId;
+     *
+     *      if (condition)
+     *          make [text, otherText] visible on each
+     * </pre>
+     * This code will change the behavior of fields <i>text</i> and <i>otherText</i> to <i>visible</i> on each transition that contains given fields.
+     * @param list of fields which behaviour will be changed
+     * @param behavior one of visible, editable, required, optional, hidden, forbidden
```

Comment:
since `initial` was introduced, it should be added to both doc lists

Reply:
initial added